### ISSUE TITLE
too many open files

### ISSUE THREAD

**Issue by shirakun** [Etosticity/proxyfilter#1]
```
root@vultr:~/proxyfilter# ./proxyfilter -file proxy.txt -output http.txt
ProxyFilter v1.3.4-alpha
Filter Bad Or Slow Proxies Out From Big Lists
Original Code In Python By @godacity_. Ported Over To Golang by @Etosticity.
Discord: https://discord.gg/4jy8khC

Proxy File:     proxy.txt
Proxy Output:   http.txt
Proxy Timeout:  2000ms

open http.txt: too many open files
```

**Comment by Etosticity** [comment]
Hi @shirakun, looks like you're running on limited resources and my program is eating it all up. An investigation is underway :)

**Comment by shirakun** [comment]
Hopefully it can custom threads
I run Vps with 512M memory in vultr

**Comment by Etosticity** [comment]
Hi @shirakun, your problem of too much resources being used at the same time should now be fixed as of commit 79a074d.

Please run your copy of this tool with this additional argument: `-routines 100`
E.g. `go run main.go -file _lists/proxy.txt -output http.txt -routines 100`

You may tweak the number of routines allowed at once depending on how fast you want the program to work 😀
